src/main/java/org/folio/services/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,142 @@
+          return;
+        }
+        asyncResultHandler.handle(Future.succeededFuture(ar.result()));
+      });
+    return this;
+  }
+
+  @Override
+  public UserService getUserTenants(String currentTenantId, String username, String userId,
+      String requestedTenantId, JsonObject headers, Handler<AsyncResult<JsonArray>> asyncResultHandler) {
+    try {
+      Map<String,String> okapiHeaders = LoginConfigUtils.decodeJsonHeaders(headers);
+      getUserTenants(okapiHeaders, currentTenantId, username, userId, requestedTenantId)
+        .onComplete(ar -> {
+          if (ar.failed()) {
+            asyncResultHandler.handle(Future.failedFuture(ar.cause()));
+            return;
+          }
+          asyncResultHandler.handle(Future.succeededFuture(ar.result()));
+        });
+    } catch (Exception ex) {
+      logger.error(ex.getMessage(), ex);
+      asyncResultHandler.handle(Future.failedFuture(ex));
+    }
+    return this;
+  }
+
+  static String buildUserLookupURL(String okapiURL, String username, String userId) {
+    String query;
+    if (username != null) {
+      query = "username==" + StringUtil.cqlEncode(username);
+    } else {
+      query = "id==" + StringUtil.cqlEncode(userId);
+    }
+    return okapiURL + "/users?query=" + PercentCodec.encode(query);
+  }
+
+  private JsonObject extractUserFromLookupResponse(HttpResponse<Buffer> res, String requestURL, String username)
+      throws LoginAPI.UserLookupException {
+    if (res.statusCode() != 200) {
+      String message = "Error looking up user at url '" + requestURL + "' Expected status code 200, got '" +
+        res.statusCode() + "' :" + res.bodyAsString();
+      throw new LoginAPI.UserLookupException(message);
+    }
+    JsonObject lookupResult = res.bodyAsJsonObject();
+    if (!lookupResult.containsKey("totalRecords") || !lookupResult.containsKey("users")) {
+      throw new LoginAPI.UserLookupException("Error, missing field(s) 'totalRecords' and/or 'users' in user response object");
+    }
+    int recordCount = lookupResult.getInteger("totalRecords");
+    if (recordCount > 1) {
+      throw new LoginAPI.UserLookupException("Bad results from username");
+    }
+    if (recordCount == 0) {
+      throw new LoginAPI.UserLookupException("No user found by username " + username);
+    }
+    return lookupResult.getJsonArray("users").getJsonObject(0);
+  }
+
+  private Future<JsonArray> getUserTenants(Map<String,String> headers, String currentTenantId, String username,
+      String userId, String requestedTenantId) {
+    Map<String,String> okapiHeaders = new CaseInsensitiveMap<>(headers);
+    String okapiUrl = okapiHeaders.get(XOkapiHeaders.URL);
+    String okapiToken = okapiHeaders.get(XOkapiHeaders.TOKEN);
+    String requestUrl = String.format(REQUEST_URL_TEMPLATE, okapiUrl, USER_TENANTS_URI_PATH);
+
+    HttpRequest<Buffer> request = WebClientFactory.getWebClient(vertx).getAbs(requestUrl);
+    if (userId != null)
```

Comment:
if userId and username are supplied, which would take precendence in mod-users?

Reply:
I both are supplied, both have to match for success. So, there is no precedence.